services/dkg/pedersen/handler.go

```diff
@@ -168,12 +170,10 @@ mainSwitch:
 			publicShares = append(publicShares, ballotShares)
 		}
 
-		h.RUnlock()
```

Comment:
DId you mean to remove the lock altogether ? No we never unlock the readlock from line 155

Reply:
oops, no that's probably an error from my side